ETL now maps BMI, Height and Smoking codes. All unit tests PASS.

shanmukhasampath opened the pull request:
This is for issue #45 

Fixed `visit_occurrence` table and it tests to properly add visit to BIOBANK which will be used to map BMI, Height and Smoking codes to respective OMOP CDM domain tables.

Fixed `stem_medical_events` table to properly process BMI, Height and Smoking codes with default domain as Measurement, Measurement and Observation.

BMI and Height codes along with their values are added to Measurement table. Unit tests are added to check standard mapping for BMI and Height codes. Unit test also includes the scenario where it rejects to add the BMI and Height codes.

Smoking codes are added to Observation table and unit tests were added to check standard mapping along with the case when they are not added into Observation table.

javier-gracia-tabuenca-tuni commented on 3_etl_code/ETL_Orchestration/sql/etl_stem_medical_events.sql:
```diff
@@ -282,10 +370,17 @@ LEFT JOIN ( SELECT SOURCE,
-   ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
-   ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
+ON CASE
+        WHEN ssfgc.SOURCE IN ("BMI","HEIGHT") THEN ssfgc.SOURCE = fgc.SOURCE AND
```
i guess you did this bcs the codes are not in service_sector_fg_codes
but I think we have to create them there 
and remove this code

shanmukhasampath replied:
Ah no Javier. The codes are in `service_sector_fg_codes` but the problem is the following SQL block
```
ssfgc.vocabulary_id = fgc.vocabulary_id AND
ssfgc.FG_CODE1 IS NOT DISTINCT FROM fgc.FG_CODE1 AND
ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
```
Here we are using `vocabulary_id` and if we do this then for the following BMI case

|SOURCE|FG_CODE1|FG_CODE2|FG_CODE3|VOCABULARY_ID|
|:----|:----|:----|:----|:----|
|BMI|NULL|NULL|NULL|FGVisitType|

This will map to every other code with no mapping an example below

|SOURCE|FG_CODE1|FG_CODE2|FG_CODE3|VOCABULARY_ID|FG_CODE5|FG_CODE|
|:----|:----|:----|:----|:----|:----|:----|
|INPAT|NULL|NULL|NULL|FGVisitType|T10|NULL|
|OUTPAT|NULL|NULL|NULL|FGVisitType|T25|NULL|
|PRIM_OUT|NULL|NULL|NULL|FGVisitType|NULL|T20|

Which is why I had to change the above code for BMI and Height codes to the following

```
ssfgc.SOURCE = fgc.SOURCE AND
ssfgc.vocabulary_id = fgc.vocabulary_id AND
ssfgc.FG_CODE1 IS NOT DISTINCT FROM fgc.FG_CODE1 AND
ssfgc.FG_CODE2 IS NOT DISTINCT FROM fgc.FG_CODE2 AND
ssfgc.FG_CODE3 IS NOT DISTINCT FROM fgc.FG_CODE3
```

javier-gracia-tabuenca-tuni commented on 3_etl_code/ETL_Orchestration/sql/etl_measurement.sql:
```diff
@@ -99,16 +105,25 @@ SELECT
 # unit_source_concept_id
-  0 AS unit_source_concept_id,
+  CASE
+       WHEN mfrwsasci.SOURCE = "HEIGHT" THEN 2001800125
```
dotn like the idea of hardcoding a concept_id code in the etl code

shanmukhasampath replied:
Yes, this one has to be fixed within in the `stem_medical_events` table as this is new.

javier-gracia-tabuenca-tuni reviewed (commented):
I dint realized that now the observation and measurement include extra info (value, unit, etc)
IMO, we need to add this into the steam table. Steam table to include value, unit, .. columns.
These are filled in in the register to steam step (mostly empty for all the registrers)
Then the steam to omop-event-table will take values from there

shanmukhasampath commented:
Added two new SQL files for measurement and observation tables which gets processed output from `finngenid_info` table.

`etl_biobank_measurement.sql` takes care of BMI and Height codes for the finngenids. `measurement_id` has offset added. Added tests to check for different scenarios and all of them pass.

`etl_biobank_observation.sql` takes care of Smoking codes for the finngenids. `observation_id` has offset added. Added tests to check for different scenarios and all of them pass.

javier-gracia-tabuenca-tuni reviewed (changes requested):
thanks, this looks so much simpler and clean now

some minor changes
 - can we rename to elt_finngenid_info_to_measurement.sql and elt_finngenid_info_to_observation.sql
 - move the tests to its own file  unittest_elt_finngenid_info_to_measurement.r and unittest_elt_finngenid_info_to_observation.r
 - inlcude the updated run_config_template.r, there should be 
 
  "elt_finngenid_info_to_measurement" , here::here("sql/elt_finngenid_info_to_measurement.sql"), here::here("tests/unittest_elt_finngenid_info_to_measurement.R"), TRUE, FALSE,
("elt_finngenid_info_to_observation" , here::here("sql/elt_finngenid_info_to_observationsql"), here::here("tests/unittest_elt_finngenid_info_to_observation.R"), TRUE, FALSE,

shanmukhasampath commented:
All the proposed changes are done

- Renamed `etl_biobank_measurement.sql` to  `etl_finngenid_info_to_measurement.sql` 
- Renamed `etl_biobank_observation.sql` to  `etl_finngenid_info_to_observation.sql`
- Created a new unit test file for `etl_finngenid_info_to_measurement.sql` as `unittest_etl_finngenid_info_to_measurement_table.R`
- Created a new unit test file for `etl_finngenid_info_to_observation.sql` as `unittest_etl_finngenid_info_to_observation_table.R`
- Added the new SQL and Unit test parts to `run_config_template.R`

All tests PASS.

javier-gracia-tabuenca-tuni reviewed (approved):
yesss, kiitos
